### PR TITLE
NISRA Individual Census

### PR DESCRIPTION
**Pull request by MebinAbraham**
### What is the context of this PR?
Added the necessary jsonnet blocks to allow the building of Northern Ireland Census schema.
After discussing with @ajmaddaford we decided that I would be best as a first pass to essentially duplicate any blocks that require a change for NIR then possibly reduce them depending on complexity vs readability of the schemas.

So for this PR, if a block required a change for NIR then a new block was created. 

The google sheets are attached on the [trello](https://trello.com/c/EXIw5C8M/2939-nisra-individual-census)  card. I have created a separate **IND_NI_DEV** sheet which you may find useful.

With Wales and England, most of the routing was the same, and it was mainly content and/or additional answer options that changed. Trying to add NIR the same way as England/Wales, led to potentially confusing and hard to follow jsonnet logic. 

[NIR Quick Launch](https://stagingv3-surveys-launch.dev.eq.ons.digital/quick-launch?url=https://gist.githubusercontent.com/MebinAbraham/a7a4fe1651a20f4378a1b6ef4a384253/raw/35044db23b13c101d1ce6e03e1ed5520d28bd65f/nisra)

### Rough figures 🤷‍♂
---

#### By duplicating for any change (current approach):
Blocks unique to ENG/WLS: **40**
Blocks unique to NIR: **28**
Blocks reused: **29**

#### By duplicating only those that affect routing (more jsonnet logic):
Blocks unique to ENG/WLS: **30**
Blocks unique to NIR: **18**
Blocks reused: **39**

#### By not duplicating (most jsonnet logic to concatenate arrays + objects based on region):
Blocks unique to ENG/WLS: **20**
Blocks unique to NIR: **8**
Blocks reused: **49**

### How to review 
- Is the NIR schema as documented in google sheets/overflow?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -91,6 +91,7 @@ requirements.txt
 
 # Ignore built schemas
 data/
+data-source/json/census_*
```
Comment (jonnyshaw89) [inline .gitignore]: The Census schemas should never be in this dir?

Comment (MebinAbraham) [reply .gitignore]: I'll revert this as previously I did add them but turned out linting breaks on census schemas.

**Comment by jonnyshaw89** [comment]
@MebinAbraham the quicklaunch is for a gist so wont be updated?

File: scripts/build_schemas.sh [hunk 2]
```diff
@@ -2,18 +2,25 @@
...
     # Lowercase the region code and replace '-' with '_'
     FORMATTED_REGION_CODE=$(echo "${region_code}" | tr '[:upper:]' '[:lower:]' | tr - _)
     DESTINATION_FILE="data-source/jsonnet/census_individual_${FORMATTED_REGION_CODE}.json"
 
-    jsonnet --tla-str region_code=${region_code} --tla-str census_date="2019-09-01" "${SOURCE_FILE}" > "${DESTINATION_FILE}"
+    if [[ "$region_code" = "GB-NIR" ]]; then
+        NI_SOURCE_FILE="data-source/jsonnet/nir/census_individual.jsonnet"
+        jsonnet --tla-str region_code=${region_code} "${NI_SOURCE_FILE}" > "${DESTINATION_FILE}"
+    else
+        NON_NI_SOURCE_FILE="data-source/jsonnet/eng-wls/census_individual.jsonnet"
```
Comment (ajmaddaford) [inline scripts/build_schemas.sh]: - `NON_NI_SOURCE_FILE` is not a good name, why not just use `SOURCE_FILE` for both?
- I think we should rename the directories to `england-wales` and `northern-ireland`

**Review by ajmaddaford — CHANGES_REQUESTED** [review]
While the duplication seems overkill for some pages, I think we should keep it for now until all the upcoming content and routing changes are done.

File: scripts/build_schemas.sh (hunk 2 above)
Comment (MebinAbraham) [reply scripts/build_schemas.sh]: Didn't have an `if` statement to begin with so made sense then but didn't refactor after. Have sorted this now. And also renamed the directories.

**Comment by MebinAbraham** [comment]
> @MebinAbraham the quicklaunch is for a gist so wont be updated?

Didn't want to commit the changes here. The gist is up to date with the PR.

File: data-source/jsonnet/northern-ireland/census_individual.jsonnet [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  legal_basis: 'Voluntary',
+  eq_id: 'census',
+  form_type: 'individual_gb_nir',
+  navigation: {
+    visible: false,
+  },
+  metadata: [
+    {
+      name: 'user_id',
+      validator: 'string',
```
Comment (jonnyshaw89) [inline data-source/jsonnet/northern-ireland/census_individual.jsonnet]: These have been changed to `"type": "string"` as part of https://github.com/ONSdigital/eq-survey-runner/pull/2076

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: data-source/jsonnet/northern-ireland/census_individual.jsonnet (hunk 3 above)
Comment (MebinAbraham) [reply data-source/jsonnet/northern-ireland/census_individual.jsonnet]: Good spot. Fixed.

**Review by jonnyshaw89 — APPROVED** [review]

**Review by ajmaddaford — APPROVED** [review]
